test/jit/test_isinstance.py

```diff
@@ -310,3 +310,12 @@ def fn(x: Any):
             x: int = 2
             fn(x)
             self.assertEqual(len(w), 0)
+
+    def test_empty_container_special_cases(self):
+        # Should not throw "Boolean value of Tensor with no values is
+        # ambiguous" error
+        torch._jit_internal.check_empty_containers(torch.Tensor([]))
```

Comment:
Thanks for adding the test case. I am curious how you came across this bug? Maybe we should that use case that leads you to this bug as a test case as well?

Reply:
@ansley ping

If that use case isn't suitable as unit test, I am OK with landing it as is too.